src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Primitives/CompositionElement.cs

```diff
@@ -2,26 +2,48 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Diagnostics;
-using Microsoft.Internal;
 
 namespace System.ComponentModel.Composition.Primitives
 {
     // Represents the ICompositionElement placeholder for an 
     // object that does not implement ICompositionElement
     [DebuggerTypeProxy(typeof(CompositionElementDebuggerProxy))]
-    internal class CompositionElement : SerializableCompositionElement
+    internal class CompositionElement : ICompositionElement
     {
-        private static readonly ICompositionElement UnknownOrigin = new SerializableCompositionElement(SR.CompositionElement_UnknownOrigin, (ICompositionElement)null);
+        private readonly string _displayName;
+        private readonly ICompositionElement _origin;
         private readonly object _underlyingObject;
 
         public CompositionElement(object underlyingObject)
-            : base(underlyingObject.ToString(), UnknownOrigin)
         {
+            _displayName = underlyingObject.ToString() ?? string.Empty;
+            _origin = new CompositionElement(SR.CompositionElement_UnknownOrigin, (ICompositionElement)null);
             _underlyingObject = underlyingObject;
         }
 
+        public CompositionElement(string displayName, ICompositionElement origin)
+        {
+            _displayName = displayName ?? string.Empty;
+            _origin = origin;
+            _underlyingObject = null;
```

Comment:
@ViktorHofer @weshaggard I'm not sure about
```
_underlyingObject = null;
```
Note: I removed SerializableCompositionElement and therefore updating the ctors this way